### ISSUE TITLE
BUG: Inconsistent train flag in PDE loss causes two divergent forward passes

### ISSUE THREAD

**Issue by zeinali72** [zeinali72/swe-pinn#50]
## Description

In `src/losses/pde.py`, lines 21 and 24 use conflicting `train` flags:

```python
U_pred = model.apply({'params': params['params']}, pde_batch, train=True)   # L21
def U_fn(pts):
    return model.apply({'params': params['params']}, pts, train=False)      # L24
```

`U_pred` (used for `SWEPhysics`) uses `train=True`, but the Jacobian is computed through `U_fn` with `train=False`. These are **two independent forward passes** that could produce different outputs.

## Impact

If any architecture uses stochastic layers or stateful layers, the PDE residual will be computed with inconsistent `U_pred` and derivatives — introducing systematic bias. Even without stochastic layers, this is a latent bug.

## Fix

Use a single forward pass with a consistent `train` flag:

```python
def U_fn(pts):
    return model.apply({'params': params['params']}, pts, train=False)

jac_U = jax.vmap(jax.jacfwd(U_fn))(pde_batch)
U_pred = U_fn(pde_batch)  # Same function, same train flag
```

## Category
**High** · Pillar: Logic Check
